### PR TITLE
Improve display of clippy lints page when JS is disabled

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
There is no point in displaying the settings menu and the filters if JS is disabled. So in this case, this PR simply hides it:

image

For the theme handling though, I need to send a fix to mdBook first. But once done, it'll look as expected (dark if system is in dark mode).

changelog: Improve clippy lints page display when JS is disabled.

r? @Alexendoo

### CODE REVIEW

**Comment by flip1995** [comment]
2 things I also noticed with the new page:
1. The link button doesn't work for me anymore (latest firefox) image

2. The indentation of code blocks is broken/removed: https://rust-lang.github.io/rust-clippy/master/index.html#cast_possible_truncation

**Comment by flip1995** [comment]
Also the link `https://rust-lang.github.io/rust-clippy/master/index.html#/cast_possible_truncation` doesn't filter the lint list anymore, nor does it automatically expand the tab of the lint.

**Comment by GuillaumeGomez** [comment]
Doesn't come from this PR, things are currently broken it seems. Urg. I'll add GUI tests after all fixes have been sent to prevent this situation from happening again.

Sending fix for the current page shortly.

**Comment by GuillaumeGomez** [comment]
All mentioned bugs were fixed. This PR can be reviewed now. :)

File: util/gh-pages/index_template.html [hunk 1]
```diff
@@ -50,14 +51,8 @@
             <h1>Clippy Lints</h1> {# #}
         </div> {# #}
 
-        <noscript> {# #}
-            <div class="alert alert-danger" role="alert"> {# #}
-                Sorry, this site only works with JavaScript! :( {# #}
-            </div> {# #}
-        </noscript> {# #}
```
Comment (Alexendoo) [inline util/gh-pages/index_template.html:57]: Let's leave some kind of message to say the filtering requires JS

File: util/gh-pages/index_template.html [hunk 2]
```diff
@@ -24,6 +24,7 @@
     <link id="styleNight" rel="stylesheet" href="https://rust-lang.github.io/mdBook/tomorrow-night.css" disabled="true"> {# #}
     <link id="styleAyu" rel="stylesheet" href="https://rust-lang.github.io/mdBook/ayu-highlight.css" disabled="true"> {# #}
     <link rel="stylesheet" href="style.css"> {# #}
+    <noscript><link rel="stylesheet" href="noscript.css"></noscript> {# #}
```
Comment (Alexendoo) [inline util/gh-pages/index_template.html:27]: We could avoid needing a new CSS file by using the `<html class="noscript">` pattern,  having `theme.js` remove the class

Comment (GuillaumeGomez) [reply util/gh-pages/index_template.html:27]: That makes the CSS bigger for everyone but as you prefer.

File: util/gh-pages/index_template.html (hunk 1 above)
Comment (GuillaumeGomez) [reply util/gh-pages/index_template.html]: Good idea!

**Comment by GuillaumeGomez** [comment]
Added a CSS class on `HTML` with JS, removed the new CSS file and added back a message in case JS is disabled.

**Review by Alexendoo — APPROVED** [review]

**Comment by Alexendoo** [comment]
👍

@bors r+

**Pull request merged**